### PR TITLE
Payment data details

### PR DESCRIPTION
**Pull request by wallaceSF**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| **Issue**         | https://mundipagg.atlassian.net/browse/MOD-979
| **What?**         | Now we saved the transaction object in the column transaction data
| **Why?**          | It was necessary for the customer

### CODE REVIEW

File: src/Kernel/Factories/ChargeFactory.php [hunk 1]
```diff
@@ -183,6 +185,14 @@ private function treatCardData(array $tranCardData, $index)
         return $tranCardData[$index];
     }
 
+    private function treatTranData(array $tranData, $index)
```
Comment (michelpl) [inline src/Kernel/Factories/ChargeFactory.php]: Temos que evitar abreviações. Para não mudar o código todo, muda no escopo dessa função e o nome dela para handleTransactionData

**Review by michelpl — CHANGES_REQUESTED** [review]

File: src/Kernel/Factories/ChargeFactory.php [hunk 2]
```diff
@@ -144,6 +144,7 @@ private function extractTransactionsFromDbData($dbData)
             $tranAcquirerMessage = explode(',', $dbData['tran_acquirer_message']);
             $tranBoletoUrl = explode(',', $dbData['tran_boleto_url']);
             $tranCardData = explode('---', $dbData['tran_card_data']);
+            $tranData = explode('---', $dbData['tran_data']);
 
             foreach ($tranId as $index => $id) {
```
Comment (michelpl) [inline src/Kernel/Factories/ChargeFactory.php:149]: Vendo o arquivo inteiro, dá pra ver que esse foreach tá gigante já. Extrai ele pra um novo método

**Review by GabrielDeveloper — APPROVED** [review]

**Comment by wallaceSF** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/mundipagg/ecommerce-module-core/pullRequest?prid=5237038&bid=17165867)

**Comment by wallaceSF** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Clones added
============
- src/Kernel/Factories/ChargeFactory.php  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/mundipagg/ecommerce-module-core/pullRequest?prid=5237038&bid=17165867)

**Comment by wallaceSF** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/mundipagg/ecommerce-module-core/pullRequest?prid=5237038&bid=17165867)

**Comment by wallaceSF** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/mundipagg/ecommerce-module-core/pullRequest?prid=5237038&bid=17165867)

**Review by michelpl — APPROVED** [review]

**Comment by wallaceSF** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/mundipagg/ecommerce-module-core/pullRequest?prid=5237038&bid=17165867)

File: src/Kernel/Factories/ChargeFactory.php (hunk 2 above)
Comment (wallaceSF) [reply src/Kernel/Factories/ChargeFactory.php:149]: Não tem como separar sem deixar mais complexo ainda o método, já que eu seria obrigado a passar todos esses itens como parâmetro.

**Pull request merged**